### PR TITLE
Taxonomy overview

### PR DESCRIPTION
**Pull request by simanadler**
Adds an overview describing the need for a taxonomy and its benefits.
Addresses the first two tasks in issue #779.  
A separate PR will address the other tasks.

### CODE REVIEW

File: site/docs/concepts/taxonomy.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+A taxonomy defines the terms and related values that need to be commonly understood and supported across the components in the system:
+
+FybrikApplication yaml - information provided about the workload and the datasets
+Fybrik manager
+Data catalog
+Data Governance Policy Manager
+Config Policy Manager
+FybrikModules
+
```
Comment (Mohammad-nassar10) [inline site/docs/concepts/taxonomy.md:13]: They appear in the same line in the "reading" version.

File: site/docs/concepts/taxonomy.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Default taxonomies are provided by fybrik, and are meant as a starting point on which to expand.
+
+## Validation Points
+
+Fybrik validates the structures and values it receives from all external components.  
+
+Webhooks are used for validation of FybrikApplication.yaml and for FybrikModule.yaml if webhooks are enabled.  If not enabled, the validation is done in the FybrikApplication controller (manager) and the FybrikModule controller.
+
+If webhooks are enabled then invalid yamls are automatically rejected and no resource is created.  However, if webhooks are turned off the resource is created and an error is indicated in its status should taxonomy validation fail.
```
Comment (Mohammad-nassar10) [inline site/docs/concepts/taxonomy.md:44]: Maybe this sentence should be rephrased.

File: site/docs/concepts/taxonomy.md (hunk 1 above)
Comment (simanadler) [reply site/docs/concepts/taxonomy.md:13]: fixed

File: site/docs/concepts/taxonomy.md (hunk 2 above)
Comment (simanadler) [reply site/docs/concepts/taxonomy.md]: Addressed.  I hope it's clearer now.

**Review by roytman — APPROVED** [review]

**Pull request merged**